src/common.jl

```diff
@@ -0,0 +1,23 @@
+module Common
+export sqerr,
+       sqerr_deriv,
+       nlogsig,
+       nlogsig_deriv,
+       heaviside,
+       sigmoid
+
+"""Squared error"""
+sqerr(p::Number, y::Number)       = (p - y)^2
+sqerr_deriv(p::Number, y::Number) = (p - y)
```

Comment:
This will technically affect the learning rate, so I will add this in so that hyperparameters are comparable across implementations (assuming that other implementations are math purists).